optionlab/utils.py

```diff
@@ -3,7 +3,8 @@
 import datetime as dt
 from datetime import timedelta
 
-from optionlab.__holidays__ import getholidays
+from holidays import country_holidays
```

Comment:
Are you using the holidays package? It's excruciatingly slow. In the first versions of the lib, I was also using this package. I chose to create a module with holiday dates because I had many performance issues with the holidays package. In fact, those holidays in __holidays__.py I obtained from the holidays package. This isn't a problem when you run the library a few times, but when you run it thousands of times (like when optimizing a strategy), using the holidays package makes everything unfeasible.

Reply:
mhm ok. well i cached it but even if you call 10,000 times with unique dates, it still only takes 3.26ms.

<img width="1262" alt="image" src="https://github.com/rgaveiga/optionlab/assets/76454594/ed500d11-63f4-4382-acaf-17eb3c50d726">

Reply:
Ok, but how does this compare to the previous implementation? Perhaps caching (which I hadn't considered at the time) could resolve this, but back then, it was approximately one order of magnitude slower. This is why I implemented __holidyas__.py (using the dates taken from holidays package). If both implementations now offer the same level of performance, I don't see any issues. However, it's crucial to note that the holiday search functionality shouldn't become a bottleneck. While it's relatively important, it's not a core feature.

Reply:
I think it's fast enough, even if slightly slower than previous. Nobody can notice changes in speed on the order of ms. It removes 9000 lines of code so I think it's a no-brainer to remove

Reply:
It's true that if one implementation takes 1 ms and the other takes 10 ms, a human being wouldn't be able to perceive the difference. But if you use the library as an engine in an optimization process or synthetic data generation that takes one hour with one implementation and ten hours with the other... then it makes a significant difference from a human perspective. We can test later with concrete examples of this.